Implemented module1 components

sanju-epam opened the pull request:
This PR will have module 1 task which contains following components 
1. Counter component
2. SearchForm Components
3. GenereSelect component

sanju-epam committed:
Implemente module1
1. Counter component
2. SearchForm Components
3. GenereSelect component

sanju-epam committed:
fixed minor bug

anooj-curator commented on mylearning-app/src/components/SearchForm.js:
```diff
@@ -0,0 +1,44 @@
+    };
+    render() {
+        return React.createElement(
+            'div',
```
Use JSX to render the element.

anooj-curator commented on mylearning-app/src/index.js:
```diff
@@ -1,15 +1,54 @@
+let counterValue = 10;
+function updateCounter(newValue) {
+  counterValue = newValue;
+  render();
```
Calling render after each update negates the principles of using React. Your child components will automatically change the data when their props are changed, so you don't have to explicitly call the render method every time you make an update. That is the beauty of React.

You'll learn this in more detail in coming topics. But for now don't call the render method on every update.

anooj-curator commented on mylearning-app/src/index.js:
```diff
@@ -1,15 +1,54 @@
+import MovieGenres from './components/GenreSelect';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
-root.render(
```
Keep the index.js file with bare min logic. Move your logic to App component instead.

anooj-curator commented on mylearning-app/src/index.js:
```diff
@@ -1,15 +1,54 @@
+
+function render() {
+  root.render(
+      React.createElement(
```
Use JSX to render the components.

anooj-curator commented on mylearning-app/src/components/GenreSelect.js:
```diff
@@ -0,0 +1,41 @@
+          <button
+            key={genre}
+            style={{
+              padding: "10px 15px",
```
This is a good approach. But a better approach would be to create multiple classes and add them as per the selection. that way you can reuse the code elsewhere.

sanju-epam replied:
I will need more details on this, as the list of genres can vary. Perhaps we could move this to configuration data instead, as creating a class for each genre might not be the most effective approach.  I may have misunderstood the requirements,

anooj-curator replied:
By classes, I mean CSS classes, not JS classes.

sanju-epam replied:
Will do it in coming commits

sanju-epam committed:
Have modified the code as per the comment
Moved create component to jsx
removed unwanted render() from index file
move few unwanted component from index

sanju-epam committed:
code fix

sanju-epam committed:
moved css code from component to class

sanju-epam committed:
Moved all the style to css as class
moved all the logic from index to App
removed use of ref

anooj-curator commented on mylearning-app/src/components/SearchForm.js:
```diff
@@ -0,0 +1,42 @@
+
+    handleKeyPress = (event) => {
+        if (event.key === 'Enter') {
+            this.handleSearch();
```
You can pass the event object to handleSearch function

anooj-curator commented on mylearning-app/src/components/SearchForm.js:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    handleSearch = () => {
+        const inputValue =  document.querySelector(".searchbox").value;
```
read the input value fro the event object

sanju-epam replied:
handlesearch will not have event all the time, it will only have the event when user click enter
